src/fsharp/service/ServiceParseTreeWalk.fs

```diff
@@ -632,7 +632,8 @@ module SyntaxTraversal =
                 let path = SyntaxNode.SynPat p :: origPath
                 match p with
                 | SynPat.Paren (p, _) -> traversePat path p
-                | SynPat.Or (p1, p2, _, _) -> [ p1; p2] |> List.tryPick (traversePat path)
+                | SynPat.As (p1, p2, _)
```

Comment:
This looked like an omission. I don't see why we wouldn't want to dive into both subpatterns here. Without it `VisitPat` is not fired for `IsInst` when the type test if paired with an `as` (which it very often is).

Reply:
completely agree with this logic